New export feature: Having `iMouse` variable causes build failure

```
[fred@pc zhecpp]$ make
Scanning dependencies of target ZHECPP
[ 50%] Building CXX object CMakeFiles/ZHECPP.dir/export.cpp.o
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp: In function ‘int main()’:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: error: no matching function for call to ‘value_ptr()’
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.hpp:65:53: note: candidate: ‘template<class genType> const typename genType::value_type* glm::value_ptr(const genType&)’
   65 |  GLM_FUNC_DECL typename genType::value_type const * value_ptr(genType const& v);
      |                                                     ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.hpp:65:53: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:11:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::vec<2, T, Q>&)’
   11 |  GLM_FUNC_QUALIFIER T const* value_ptr(vec<2, T, Q> const& v)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:11:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:17:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::vec<2, T, Q>&)’
   17 |  GLM_FUNC_QUALIFIER T* value_ptr(vec<2, T, Q>& v)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:17:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:23:31: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::vec<3, T, Q>&)’
   23 |  GLM_FUNC_QUALIFIER T const * value_ptr(vec<3, T, Q> const& v)
      |                               ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:23:31: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:29:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::vec<3, T, Q>&)’
   29 |  GLM_FUNC_QUALIFIER T* value_ptr(vec<3, T, Q>& v)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:29:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:35:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::vec<4, T, Q>&)’
   35 |  GLM_FUNC_QUALIFIER T const* value_ptr(vec<4, T, Q> const& v)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:35:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:41:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::vec<4, T, Q>&)’
   41 |  GLM_FUNC_QUALIFIER T* value_ptr(vec<4, T, Q>& v)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:41:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:47:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<2, 2, T, Q>&)’
   47 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<2, 2, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:47:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:53:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<2, 2, T, Q>&)’
   53 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<2, 2, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:53:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:59:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<3, 3, T, Q>&)’
   59 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<3, 3, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:59:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:65:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<3, 3, T, Q>&)’
   65 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<3, 3, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:65:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:71:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<4, 4, T, Q>&)’
   71 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<4, 4, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:71:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:77:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<4, 4, T, Q>&)’
   77 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<4, 4, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:77:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:83:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<2, 3, T, Q>&)’
   83 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<2, 3, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:83:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:89:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<2, 3, T, Q>&)’
   89 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<2, 3, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:89:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:95:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<3, 2, T, Q>&)’
   95 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<3, 2, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:95:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:101:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<3, 2, T, Q>&)’
  101 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<3, 2, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:101:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:107:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<2, 4, T, Q>&)’
  107 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<2, 4, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:107:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:113:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<2, 4, T, Q>&)’
  113 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<2, 4, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:113:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:119:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<4, 2, T, Q>&)’
  119 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<4, 2, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:119:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:125:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<4, 2, T, Q>&)’
  125 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<4, 2, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:125:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:131:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<3, 4, T, Q>&)’
  131 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<3, 4, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:131:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:137:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<3, 4, T, Q>&)’
  137 |  GLM_FUNC_QUALIFIER T* value_ptr(mat<3, 4, T, Q>& m)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:137:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:143:30: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::mat<4, 3, T, Q>&)’
  143 |  GLM_FUNC_QUALIFIER T const* value_ptr(mat<4, 3, T, Q> const& m)
      |                              ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:143:30: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:149:25: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::mat<4, 3, T, Q>&)’
  149 |  GLM_FUNC_QUALIFIER T * value_ptr(mat<4, 3, T, Q>& m)
      |                         ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:149:25: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:155:31: note: candidate: ‘template<class T, glm::qualifier Q> const T* glm::value_ptr(const glm::qua<T, Q>&)’
  155 |  GLM_FUNC_QUALIFIER T const * value_ptr(qua<T, Q> const& q)
      |                               ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:155:31: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

In file included from /usr/include/glm/gtc/type_ptr.hpp:230,
                 from /home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:14:
/usr/include/glm/gtc/type_ptr.inl:161:24: note: candidate: ‘template<class T, glm::qualifier Q> T* glm::value_ptr(glm::qua<T, Q>&)’
  161 |  GLM_FUNC_QUALIFIER T* value_ptr(qua<T, Q>& q)
      |                        ^~~~~~~~~
/usr/include/glm/gtc/type_ptr.inl:161:24: note:   template argument deduction/substitution failed:
/home/fred/Workspace/SHADERed/bin/zhecpp/export.cpp:417:77: note:   candidate expects 1 argument, 0 provided
  417 | formLocation(Simple_SP, "iMouse"), 1, glm::value_ptr());
      |                                                      ^

make[2]: *** [CMakeFiles/ZHECPP.dir/build.make:63: CMakeFiles/ZHECPP.dir/export.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/ZHECPP.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
[fred@pc zhecpp]$ cd ..
```

Removing the variable fixes the issue.

--- a/Objects/Export/ExportCPP.cpp
+++ b/Objects/Export/ExportCPP.cpp
@@ -185,6 +185,7 @@ namespace ed
 		case ed::SystemShaderVariable::GeometryTransform: return "sysGeometryTransform";
 		case ed::SystemShaderVariable::ViewProjection: return "sysViewProjection";
 		case ed::SystemShaderVariable::ViewOrthographic: return "sysViewOrthographic";
+		case ed::SystemShaderVariable::MousePosition: return "sysMousePosition";
 		}
 
 		return "";
@@ -454,6 +455,8 @@ namespace ed
 			initSrc += indent + "// system variables\n";
 			initSrc += indent + "float sysTime = 0.0f, sysTimeDelta = 0.0f;\n";
 			initSrc += indent + "unsigned int sysFrameIndex = 0;\n";
+
+			initSrc += indent + "glm::vec2 sysMousePosition(sedMouseX, sedMouseY);\n";
 			initSrc += indent + "glm::vec2 sysViewportSize(sedWindowWidth, sedWindowHeight);\n";
 			initSrc += indent + "glm::mat4 sysView(";
 			for (int i = 0; i < 16; i++) {

--- a/bin/data/export/cpp/template.cpp
+++ b/bin/data/export/cpp/template.cpp
@@ -66,6 +66,7 @@ int main()
 
 	// required:
 	float sedWindowWidth = 800, sedWindowHeight = 600;
+	float sedMouseX = 0, sedMouseY = 0;
 
 	std::chrono::time_point<std::chrono::system_clock> timerStart = std::chrono::system_clock::now();
 
@@ -77,11 +78,15 @@ int main()
 	while (run) {
 		while (SDL_PollEvent(&event))
 		{
-			if (event.type == SDL_QUIT)
+			if (event.type == SDL_QUIT) {
 				run = false;
-			else if (event.type == SDL_WINDOWEVENT && event.window.event == SDL_WINDOWEVENT_RESIZED) {
+			} else if (event.type == SDL_MOUSEMOTION) {
+				sedMouseX = event.motion.x;
+				sedMouseY = event.motion.y;
+				sysMousePosition = glm::vec2(sedMouseX/sedWindowWidth, 1.f-(sedMouseY/sedWindowHeight));
+			} else if (event.type == SDL_WINDOWEVENT && event.window.event == SDL_WINDOWEVENT_RESIZED) {
 				sedWindowWidth = event.window.data1;
-                sedWindowHeight = event.window.data2;
+				sedWindowHeight = event.window.data2;
 
 				[$$resize_event$$]
 			}

